common/djangoapps/enrollment/views.py

```diff
@@ -360,11 +362,12 @@ def post(self, request):
         # Check whether any country access rules block the user from enrollment
         # We do this at the view level (rather than the Python API level)
         # because this check requires information about the HTTP request.
+
+        # Lookup the user, instead of using request.user, since request.user may not match the username POSTed.
+        user = User.objects.get(username=username)
```

Comment:
What if the user object doesn't exist when `has_api_key_permissions=True`?  Should we catch `DoesNotExist` here?